### PR TITLE
Don't include Android model by default

### PR DESCRIPTION
**Pull request by pilgrim-brave**
Resolves https://github.com/brave/brave-browser/issues/12638

## Submitter Checklist:

- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [x] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [x] Ran `git rebase master` (if needed).
- [ ] Requested a security/privacy review as needed.

## Reviewer Checklist:

- [ ] New files have MPL-2.0 license header.
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on.
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

### CODE REVIEW

**Comment by mkarolin** [comment]
Are we intentionally only changing this in `BuildUserAgentFromProduct`, but not in `BuildUserAgentFromProductAndExtraOSInfo`? 

If we do want to change it in both places, then there is a bit more convoluted way that avoids patching. 

We could use `chromium_src` override to redefine `HardwareModelName` (in `base::SysInfo::HardwareModelName`, which is what ultimately gets the model name - called from `BuildModelInfo`) with `EmptyHardwareModelName`. Then use `chromium_src` overrides for `base/system/sys_info.h` and `base/system/sys_info_android.cc` to add that new method.

**Comment by pilgrim-brave** [comment]
The thing is that we DO want the model to show in brave://version so we can't patch any "lower" than the functions that construct the user agent string.

**Comment by mkarolin** [comment]
I see, then we could replace the entire `BuildUserAgentFromProduct` function definition in `chromium_src` override. Though, I would be a bit reluctant to do that without at least some kind of unit test to check that we don't diverge from Chromium's original in this function. WDYT, @bridiver? 

If that's undesirable, we could maybe use `chromium_src` override 
<pre>
#define BRAVE_USER_AGENT true ? IncludeAndroidModel::Exclude :
</pre>
to patch after line 234 instead of doing a replacement patch.

File: patches/content-common-user_agent.cc.patch [hunk 1]
```diff
@@ -0,0 +1,13 @@
+diff --git a/content/common/user_agent.cc b/content/common/user_agent.cc
+index a87ca565f8aa809a46086861c7057da013e9c911..ab271cb3020eb79fa34cd9d0c675aaecd01faad3 100644
+--- a/content/common/user_agent.cc
++++ b/content/common/user_agent.cc
+@@ -233,7 +233,7 @@ std::string BuildUserAgentFromProduct(const std::string& product) {
+   std::string os_info;
+   base::StringAppendF(&os_info, "%s%s", GetUserAgentPlatform().c_str(),
+                       BuildOSCpuInfo(IncludeAndroidBuildNumber::Exclude,
+-                                     IncludeAndroidModel::Include)
```
Comment (bridiver) [inline patches/content-common-user_agent.cc.patch:9]: would it maybe be better to override `HardwareModelName` to some fixed value?

**Review by bridiver — APPROVED** [review]

**Comment by pilgrim-brave** [comment]
CI failures are unrelated.

**Pull request merged**